From 1926699b02882bc5ed659d4debce2a0127d2edf0 Mon Sep 17 00:00:00 2001
From: Alexandre Derumier <aderumier@odiso.com>
Date: Wed, 5 Jun 2019 14:47:05 +0200
Subject: [PATCH 6/7] ifreload: down/up vxlan && vlan interfaces when
 ifreload_down_changed=0

almost all attributes of vxlan interfaces can't be updated
(including vxlan-id).

Same for vlan interfaces (vlan-protocol, vlan-id)

so when ifreload_down_changed=0, ifreload can't update vxlan or vlan
 attributes.

fix: https://github.com/CumulusNetworks/ifupdown2/issues/50
Signed-off-by: Alexandre Derumier <aderumier@odiso.com>
Signed-off-by: Thomas Lamprecht <t.lamprecht@proxmox.com>
---
 ifupdown2/ifupdown/ifupdownmain.py | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/ifupdown2/ifupdown/ifupdownmain.py b/ifupdown2/ifupdown/ifupdownmain.py
index b73c742..65fd67e 100644
--- a/ifupdown2/ifupdown/ifupdownmain.py
+++ b/ifupdown2/ifupdown/ifupdownmain.py
@@ -2355,7 +2355,9 @@ class ifupdownMain:
                                      % (newifaceobjlist[objidx].name,
                                         ifaceLinkKind.to_str(lastifaceobjlist[0].link_kind)))
                     ifacedownlist.append(newifaceobjlist[objidx].name)
-                if not down_changed:
+
+                ifacelink_kind = ifaceLinkKind.to_str(lastifaceobjlist[0].link_kind)
+                if not down_changed and ifacelink_kind != 'vxlan' and ifacelink_kind != 'vlan':
                     continue
                 if len(newifaceobjlist) != len(lastifaceobjlist):
                     ifacedownlist.append(ifname)
-- 
2.30.2

